src/Cli/Microsoft.TemplateEngine.Cli/Commands/InteractiveMode/InteractiveMode.cs

```diff
@@ -0,0 +1,253 @@
+
+        internal static bool ShouldEnterInteractiveMode(ParseResult parsedArgs)
+        {
+            // TODO: find a better way to query for the option
+            // TODO: Better handling of this option when executing
+            bool interactiveOptionValue = false;
+            if (NewCommand.InteractiveTemplateOption is not null)
+            {
+                interactiveOptionValue = parsedArgs.GetValue(NewCommand.InteractiveTemplateOption);
+            }
+
+            var envVariable = Environment.GetEnvironmentVariable("DOTNETNEWINTERACTIVE");
+            bool envVariableValue = Convert.ToBoolean(envVariable);
+
+            if ((interactiveOptionValue || envVariableValue) && DoesSupportInteractive())
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        internal static bool DoesSupportInteractive()
+        {
+            // hard coded for now
+            // Check if current environment supports what we want to do (If we need any fancy stuff)
+            return true;
+        }
+
+        internal static TemplateCommand? GetTemplate(
+            InstantiateCommandArgs args,
+            IEngineEnvironmentSettings environmentSettings,
+            IEnumerable<TemplateGroup> templateGroups,
+            TemplatePackageManager templatePackageManager)
+        {
+            TemplateConstraintManager constraintManager = new(environmentSettings);
+
+            foreach (TemplateGroup templateGroup in templateGroups.Where(template => template.ShortNames.Contains(args.ShortName)))
+            {
+                foreach (IGrouping<int, CliTemplateInfo> templateGrouping in GetAllowedTemplates(constraintManager, templateGroup).GroupBy(g => g.Precedence).OrderByDescending(g => g.Key))
+                {
+                    foreach (CliTemplateInfo template in templateGrouping)
+                    {
+                        // Make some additional check based on user input match (like template language)
+                        try
+                        {
+                            TemplateCommand command = new(
+                                args.Command,
+                                environmentSettings,
+                                templatePackageManager,
+                                templateGroup,
+                                template);
+
+                            // simplify this for now to just return one template
+                            return command;
+                        }
+                        catch (InvalidTemplateParametersException e)
+                        {
+                            Console.Error.WriteLine(LocalizableStrings.GenericWarning, e.Message);
+                        }
+                    }
+                }
+            }
+            return null;
+        }
+
+        private static IEnumerable<CliTemplateInfo> GetAllowedTemplates(TemplateConstraintManager constraintManager, TemplateGroup templateGroup)
```

Comment:
if calling method is moved to `InstantiateCommand` it should be possible to reuse those methods.